airflow/providers/amazon/aws/executors/ecs/ecs_executor.py

```diff
@@ -302,14 +302,15 @@ def __handle_failed_task(self, task_arn: str, reason: str):
                 self.__class__.MAX_RUN_TASK_ATTEMPTS,
                 task_arn,
             )
-            self.active_workers.increment_failure_count(task_key)
+            if increment:
+                self.active_workers.increment_failure_count(task_key)
             self.pending_tasks.appendleft(
                 EcsQueuedTask(
                     task_key,
                     task_cmd,
                     queue,
                     exec_info,
-                    failure_count + 1,
+                    failure_count + 1 if increment else failure_count,
```

Comment:
Why do we increment the failure count twice?

Reply:
I guess I'm just confused (or maybe forgetting) why we're keeping track of that thing in two separate places?

Reply:
I think it was just for convenience of not having `self.active_workers.failure_count_by_key(task_key)` six times.   How do you feel about this alternative?  I kinda like this, and unit tests like it too:

![image](https://github.com/apache/airflow/assets/1920178/17bf082f-8f62-4797-aa23-4557da29151e)

Reply:
as a helper

![image](https://github.com/apache/airflow/assets/1920178/59fc0e62-4234-4681-9e81-61c13153269c)

Reply:
I like the idea of a helper function!

Reply:
Sure.  I'll make it happen.